sum.go

```diff
@@ -115,6 +112,32 @@ func isBlake2b(code uint64) bool {
 	return code >= BLAKE2B_MIN && code <= BLAKE2B_MAX
 }
 
+func sumBlake2b256(data []byte) []byte {
+	out := blake2b.Sum256(data)
+	return out[:]
+}
+
+const blake2b384Size = 384 / 8
+
+var blake2b384Config = &blake2b.Config{Size: blake2b384Size}
+
+func sumBlake2b384(data []byte) []byte {
+	hasher, err := blake2b.New(blake2b384Config)
+	if err != nil {
+		panic(err)
+	}
+	hasher.Write(data)
+	var out [blake2b384Size]byte
+	copy(out[:], hasher.Sum(nil)[:blake2b384Size])
```

Comment:
Why the reslice of the destination here?

Reply:
I was worried that the hasher would over allocate. However, this turns out to not be the case so I'll correct it.